### PR TITLE
Process cli arg

### PR DESCRIPTION
**Pull request by gbataille**

### CODE REVIEW

File: app/Main.hs [hunk 1]
```diff
@@ -1,3 +1,27 @@
...
+
+import Data.Text (unpack)
+
+import Krank
+import Krank.Formatter
+
 main :: IO ()
 main = do
-  putStrLn "krank is cool"
+  filePath <- parseArgs
```
Comment (guibou) [inline app/Main.hs]: Please use `optparse-applicative` instead: http://hackage.haskell.org/package/optparse-applicative

In the future we'll have a more complex command line and it comes with a really great interface for type safe cli interface (with help and auto-completion generation).

File: src/Krank/Formatter.hs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+import Krank.Types
+
+showViolations :: [Violation]
+               -> Text
+showViolations = foldl (\acc v -> acc `append` showViolation v) ""
+
+showViolation :: Violation
+              -> Text
+showViolation violation = "\n"
```
Comment (guibou) [inline src/Krank/Formatter.hs]: I'd like to use `PyF` for this kind of formatting.

File: src/Krank/Types.hs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+module Krank.Types (
+  Violation(..)
+  ) where
+
+import Data.Text (Text)
+
+data Violation = Violation { checker :: Text
+                           , level :: Int
```
Comment (guibou) [inline src/Krank/Types.hs]: Can we imagine a `data = Warning | Info | Foo` for `level`?

File: src/Krank/Formatter.hs [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  showViolations
+  ) where
+
+import Data.Text (Text, append, pack)
+
+import Krank.Types
+
+showViolations :: [Violation]
+               -> Text
+showViolations = foldl (\acc v -> acc `append` showViolation v) ""
```
Comment (guibou) [inline src/Krank/Formatter.hs]: nit: `foldMap` is doing exactly that.

`foldMap showViolation`

**Review by guibou — CHANGES_REQUESTED** [review]
A few comments about the style.

One mandatory request: use `optaparse-applicative` for command line argument reading.

File: app/Main.hs (hunk 1 above)
Comment (gbataille) [reply app/Main.hs]: I don't doubt that future would need something like that :)
I just felt like starting simple. I'll look into it

File: src/Krank/Formatter.hs (hunk 2 above)
Comment (gbataille) [reply src/Krank/Formatter.hs]: sounds good to me. I'll look into it

File: src/Krank/Types.hs (hunk 3 above)
Comment (gbataille) [reply src/Krank/Types.hs]: yes. I was actually not happy with this `Int` I drafted once came time to display it.
I'll work on that

File: src/Krank/Formatter.hs (hunk 4 above)
Comment (gbataille) [reply src/Krank/Formatter.hs]: And one new tool in my toolbelt! thanks.

Comment (guibou) [reply src/Krank/Formatter.hs]: Note that it works because text is a `Semigroup` (also a `Monoid`).

File: app/Main.hs [hunk 5]
```diff
@@ -1,27 +1,32 @@
...
 
 import Krank
 import Krank.Formatter
 
-main :: IO ()
-main = do
-  filePath <- parseArgs
-  case filePath of Just f -> putStrLn . unpack . showViolations $ processFile f
-                   Nothing -> printHelp >> exitFailure
+data Sample = Sample {
```
Comment (guibou) [inline app/Main.hs]: Nitpick: is `Sample` an adapted name?

File: app/Main.hs [hunk 6]
```diff
@@ -1,27 +1,32 @@
...
+fileToParse = Opt.argument Opt.str (Opt.metavar "FILE")
 
-parseArgs :: IO (Maybe FilePath)
-parseArgs = do
-  args <- getArgs
-  case args of [a] -> pure $ Just a
-               _   -> pure Nothing
+sample :: Opt.Parser Sample
+sample = Sample
+  <$> fileToParse
```
Comment (guibou) [inline app/Main.hs]: Question of style, I'd prefer if `fileToParse` was inlined here.

**Review by guibou — APPROVED** [review]
I left some style comments, but I totally approve!

File: app/Main.hs (hunk 5 above)
Comment (gbataille) [reply app/Main.hs]: :D copy/paste from the doc. changing

File: app/Main.hs (hunk 6 above)
Comment (gbataille) [reply app/Main.hs]: You mean writing the statement on a single line?

Comment (guibou) [reply app/Main.hs]: Yes.

Comment (guibou) [reply app/Main.hs]: No. ;)

Comment (guibou) [reply app/Main.hs]: I meant:

```
sample = Sample 
   <$> the_code_of_file_to_parse_directly_here.
```

**Pull request merged**